django/src/rdwatch/tasks/__init__.py

```diff
@@ -1,6 +1,275 @@
+import io
+import logging
+import os
+from datetime import datetime, timedelta
+
 from celery import shared_task
+from PIL import Image
+from pyproj import Transformer
+
+from django.core.files import File
+
+from rdwatch.models import SiteImage, SiteObservation
+from rdwatch.utils.raster_tile import get_raster_bbox
+from rdwatch.utils.satellite_bands import get_bands
+from rdwatch.utils.worldview_processed.raster_tile import (
+    get_worldview_processed_visual_bbox,
+)
+from rdwatch.utils.worldview_processed.satellite_captures import (
+    get_captures as get_worldview_captures,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def scale_bbox(bbox: tuple[float, float, float, float], scale_factor: float):
+    xmin, ymin, xmax, ymax = bbox
+    width = xmax - xmin
+    height = ymax - ymin
+    new_width = width * scale_factor
+    new_height = height * scale_factor
+    new_xmin = xmin - ((new_width - width) / 2)
+    new_ymin = ymin - ((new_height - height) / 2)
+    new_xmax = xmax + ((new_width - width) / 2)
+    new_ymax = ymax + ((new_height - height) / 2)
+    scaled_bbox = [new_xmin, new_ymin, new_xmax, new_ymax]
+    return scaled_bbox
+
+
+def get_max_bbox(
+    bbox: tuple[float, float, float, float], maxbbox: tuple[float, float, float, float]
+):
+    newbbox = [
+        min(bbox[0], maxbbox[0]),
+        min(bbox[1], maxbbox[1]),
+        max(bbox[2], maxbbox[3]),
+        max(bbox[2], maxbbox[3]),
+    ]
+    return newbbox
+
+
+def get_percent_black_pixels(bytes: bytes):
+    img = Image.open(io.BytesIO(bytes))
+    # determine the size of the image
+    width, height = img.size
+
+    # get the pixel data of the image
+    pixels = img.load()
+
+    # count the number of black pixels in the image
+    num_black_pixels = 0
+    for x in range(width):
+        for y in range(height):
+            if pixels[x, y] == (0, 0, 0):
+                num_black_pixels += 1
+
+    # calculate the percentage of black pixels in the image
+    percent_black_pixels = (num_black_pixels / (width * height)) * 100
+    return percent_black_pixels
+
+
+def get_closest_capture(
+    bbox: tuple[float, float, float, float],
+    timestamp: datetime,
+    constellation: str,
+    worldView=False,
+):
+    timebuffer = timedelta(days=1)
+    captures = None
```

Comment:
This initial assignment isn't necessary, as `if`/`elif`/`else` do not begin a new name scope. Additionally, this will make the type of the variable `list[WorldViewProcessedCapture] | None` instead of `list[WorldViewProcessedCapture]`, which would prevent a type-checker from catching certain type errors if we were using one.